### PR TITLE
xds: gate HttpFilter parsing by env flag

### PR DESCRIPTION
**Pull request by dapengzhang0**
New features should be env flag protected even for proto parsing, because otherwise if there is a bug in parsing the new fields, users will break even if they are not using new feature.

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java [hunk 1]
```diff
@@ -330,6 +332,7 @@ public void ldsResourceUpdated() {
 
   @Test
   public void ldsResourceUpdate_withFaultInjection() {
+    Assume.assumeTrue(useProtocolV3());
```
Comment (voidzcy) [inline xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:335]: Why this?

Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:335]: It does not make sense to test `ldsResourceUpdate_withFaultInjection ` with `ClientXdsClientV2Test`, the assertions are not true for v2.

**Review by voidzcy — APPROVED** [review]

File: xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java [hunk 2]
```diff
@@ -172,6 +173,7 @@ public boolean shouldAccept(Runnable command) {
 
   @Before
   public void setUp() throws IOException {
+    ClientXdsClient.enableFaultInjection = true;
```
Comment (voidzcy) [inline xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:176]: Wait, this is lazy. You got to do it correctly.

Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:176]: This is following the same style as `enableTimeout ` https://github.com/grpc/grpc-java/blob/v1.35.0/xds/src/test/java/io/grpc/xds/XdsNameResolverTest.java#L117

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:176]: I mean you need to turn it back, and cover the default/unset behavior.

Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:178]: Exactly I mean `enableTimeout` was not turned back.

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/ClientXdsClientTestBase.java:178]: Oh, really? Ah, that's bad....

**Pull request merged**